via/services/pdf_url.py

```diff
@@ -73,20 +75,35 @@ def get_pdf_url(self, url):
             return self._google_file_url(file_details, url)
 
         if self._is_onedrive_url(url):
-            return self._proxy_onedrive_pdf(url)
+            return self._proxy_python_pdf(url, route="proxy_onedrive_pdf")
+
+        if self._is_d2l_url(url):
```

Comment:
I briefly tried the nginx approach but not sure if it was possible to:

- Pass the headers to nginx in a secure way
- Do that in a way that involved sane code.